trieste/models/model_interfaces.py

```diff
@@ -71,6 +71,20 @@ def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
         """
         raise NotImplementedError
 
+    def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        """
+        Return the mean and variance of the independent marginal distributions at each point in
+        ``query_points`` for the actual observations, including noise contributions.
+
+        Note that this is not supported by all models.
+
+        :param query_points: The points at which to make predictions, of shape [..., D].
+        :return: The mean and variance of the independent marginal distributions at each point in
+            ``query_points``. For a predictive distribution with event shape E, the mean and
+            variance will both have shape [...] + E.
+        """
+        raise NotImplementedError(f"Model {self!r} does not support predicting actual observations")
```

Comment:
Model {self!r} does not support predicting actual observations, just the latent function.